script.js

```diff
@@ -0,0 +1,40 @@
+// Ensure the DOM is fully loaded before running the script
+document.addEventListener('DOMContentLoaded', function() {
+    // Add event listener to the form
+    document.getElementById('passwordForm').addEventListener('submit', function(event) {
+        event.preventDefault(); // Prevent form submission
+
+        // Retrieve the input values
+        const serviceName = document.getElementById('service-name').value.trim();
+        const username = document.getElementById('username').value.trim();
+        const password = document.getElementById('password').value.trim();
+
+        // Validation: Check if all fields are filled
+        if (!serviceName || !username || !password) {
+            alert('Please fill in all fields.');
+            return;
+        }
+
+        // Username validation: minimum length
+        if (username.length < 3) {
+            alert('Username must be at least 3 characters long.');
+            return;
+        }
+
+        // Password validation pattern
+        const passwordPattern = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/;
+
+        if (!passwordPattern.test(password)) {
+            alert('Password must contain at least one uppercase letter, one lowercase letter, one digit, and one special character.');
```

Comment:
Nice usage of regex for password validation. However, I noticed that the error message could be a bit more descriptive. Currently, it doesn’t mention the requirement for a minimum length of 8 characters. You don't need to modify it in this PR but just mentioning for learning purpose.